backup/backup.go

```diff
@@ -0,0 +1,123 @@
+package backup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	help "github.com/featureform/helpers"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"io/ioutil"
+	"time"
+)
+
+type ProviderType string
+
+const (
+	AZURE ProviderType = "AZURE"
+	LOCAL              = "LOCAL"
+)
+
+// Client Allows ETCD to be tested with mock values
+type Client interface {
+	Get(ctx context.Context, key string, opts ...clientv3.OpOption) (*clientv3.GetResponse, error)
+}
+
+type Backup struct {
+	ETCDClient   Client
+	ProviderType ProviderType
```

Comment:
turn this into Provider Provider

The Init method of both could do the env variable fill in if needed.

Reply:
Was unsure what you meant. Changed it to provider.FilestoreType so it matches with the filestores we have implemented